### ISSUE TITLE
error about training——

### ISSUE THREAD

**Issue by zhangxin253675** [Peterande/D-FINE#323]
您好，感谢你们出色的工作！
我之前已经在VisDrone-DET数据集上复现了你们的代码，已经取得较好的精度。事实上，我在复现你们的代码时，使用命令为
`CUDA_VISIBLE_DEVICES=0,1,2,3 python3 -m torch.distributed.run --nproc_per_node=4 train.py -c configs/dfine/custom/dfine_hgnetv2_s_custom.yml --use-amp --seed=0 -t ./pretrained_weight/dfine_s_obj365.pth`
，加载了预训练权重，在训练几轮后，经常（不固定在哪轮）出现如下错误：

```
# 在./D-FINE-master/src/zoo/dfine/matcher.py文件中的HungarianMatcher类中：
# Compute the giou cost betwen boxes
cost_giou = -generalized_box_iou(box_cxcywh_to_xyxy(out_bbox), box_cxcywh_to_xyxy(tgt_bbox))

# box_cxcywh_to_xyxy方法报错 assert (boxes1[:, 2:] >= boxes1[:, :2]).all()
# 这说明预测框的w或h不正确
```

这表明，D-FINE方法面对小目标较多的训练集的训练稳定性不够高，对吗？请问作者您有什么可以解决的思路吗？
非常感谢！

**Comment by Peterande** [comment]
这说明数据集里出现框的右下角坐标大于了左上角坐标，属于标签错误，可以清洗一下数据集标签。
顺便安利一篇师弟带飞的小目标检测工作：https://arxiv.org/abs/2505.05741

**Comment by zhangxin253675** [comment]
您好👋，非常感谢您的回复！

事实上，我已经对gt文件进行了坐标检验并剔除了不符合坐标要求的gt框。而且报错显示的是out_box（预测框）的坐标不满足要求。

我查了一下，csdn上有一个类似的帖子，做法是：在训练时将混合精度训练关闭。

我去掉混合精度训练设置后，进行了多次训练，发现训练时不会报错了。

所以，可能是混合精度训练导致训练不稳定，出现预测框不满足坐标要求。

感谢您出色的工作与您的回复！

---- Replied Message ----
| From | Yansong ***@***.***> |
| Date | 08/08/2025 02:07 |
| To | Peterande/D-FINE ***@***.***> |
| Cc | Zhang ***@***.***>,
Author ***@***.***> |
| Subject | Re: [Peterande/D-FINE] error about training—— (Issue #323) |
Peterande left a comment (Peterande/D-FINE#323)

这说明数据集里出现框的右下角坐标大于了左上角坐标，属于标签错误，可以清洗一下数据集标签。
顺便安利一篇师弟带飞的小目标检测工作：https://arxiv.org/abs/2505.05741

**Comment by Peterande** [comment]
看来是NaN的问题，bf16混合精度训练应该也可以避免，但不确定会不会影响最终性能～如果仍然想使用fp16，可以手动找一下数值溢出的位置clamp一下
